docs/tutorial.md

```diff
@@ -0,0 +1,285 @@
+
+```yaml
+resources:
+  Resources:
+    CustomerTable:
+      Type: AWS::DynamoDB::Table
+      Properties:
+        AttributeDefinitions:
+          - AttributeName: primary_key
+            AttributeType: S
+        BillingMode: PAY_PER_REQUEST
+        KeySchema:
+          - AttributeName: primary_key
+            KeyType: HASH
+        TableName: ${self:service}-customerTable-${sls:stage}
+```
+
+And lets create a new file in the same folder as the `serverless.yml` called `createCustomer.js` and add the following code to it:
+
+```javascript
+'use strict';
+const AWS = require('aws-sdk');
+
+module.exports.createCustomer = async (event) => {
+  const body = JSON.parse(Buffer.from(event.body, 'base64').toString());
+  const dynamoDb = new AWS.DynamoDB.DocumentClient();
+  const putParams = {
+    TableName: process.env.DYNAMODB_CUSTOMER_TABLE,
+    Item: {
+      primary_key: body.name,
+      email: body.email,
+    },
+  };
+  await dynamoDb.put(putParams).promise();
+
+  return {
+    statusCode: 201,
+  };
+};
+```
+
+You may have noticed we include an npm module to help us talk to AWS, so lets make sure we install this required npm module as a part of our service with the following command:
+
+```bash
+npm install –save aws-sdk
+```
+
+Note: If you would like this entire project as a reference to clone, you can find this on [GitHub](https://github.com/serverless/tutorial/tree/main/getting-started) but just remember to add your own org and app names to `serverless.yml` to connect to your Serverless Dashboard account before deploying.
+
+## Making the database table name available to the function
+
+In order for our function to know what table to access, we need some way to make that name available and thankfully Lambda has the concept of environment variables. You can set an environment variable in your `serverless.yml` that is then accessible to the function in code. Under the provider section of your `serverless.yml` add the following:
+
+```yaml
+provider:
+  environment:
+    DYNAMODB_CUSTOMER_TABLE: ${self:service}-customerTable-${sls:stage}
+```
+
+In our function code, you may have noticed we access this environment variable with the following: `process.env.DYNAMODB_CUSTOMER_TABLE`. Environment variables become a very powerful way to pass configuration details we need to our Lambda functions.
+
+## Setting function permissions
+
+While we could go ahead and deploy our changes already (feel free to do so with the command `serverless deploy`), we do need to add one more thing to allow our code to talk to our database. By default, and for good security reasons, AWS requires that we add explicit permissions to allow Lambda functions to access other AWS services. This requires us adding some more configuration to our `serverless.yml`. Within the `provider` block of our `serverless.yml`, make sure you have the following:
+
+```yaml
+provider:
+  iamRoleStatements:
```

Comment:
Let's not promote syntax that is no longer supported - it should be using `iam.role.statements`